controllers/ingress/astartedefaultingress_controller.go

```diff
@@ -111,5 +114,33 @@ func (r *AstarteDefaultIngressReconciler) SetupWithManager(mgr ctrl.Manager) err
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ingressv1alpha1.AstarteDefaultIngress{}, builder.WithPredicates(pred)).
 		Owns(&networkingv1.Ingress{}).
+		Watches(
+			&source.Kind{Type: &apiv1alpha1.Astarte{}},
+			handler.EnqueueRequestsFromMapFunc(astarteToADIReconcileRequestFn(r.Client)),
+		).
 		Complete(r)
 }
+
+func astarteToADIReconcileRequestFn(c client.Client) func(obj client.Object) []reconcile.Request {
+	return func(obj client.Object) []reconcile.Request {
+		astarteName := obj.GetName()
+		ret := []reconcile.Request{}
+		adiList := &ingressv1alpha1.AstarteDefaultIngressList{}
+		_ = c.List(context.Background(), adiList, client.InNamespace(obj.GetNamespace()))
+
+		// we assume only one AstarteDefaultIngress is present
+		if len(adiList.Items) > 0 {
+			adi := adiList.Items[0]
```

Comment:
What if, instead of assuming this, you do a `for`?

Reply:
Even if having multiple ADI resources is not the standard approach, it is legitimate indeed. Going with the `for`.